modules/python/python-source.c

```diff
@@ -276,12 +288,39 @@ _py_lookup_suspend_and_wakeup_methods(PythonSourceDriver *self)
   return TRUE;
 }
 
+static gboolean
+_py_lookup_generate_persist_name_method(PythonSourceDriver *self)
+{
+  self->py.generate_persist_name = _py_get_attr_or_null(self->py.instance, "generate_persist_name");
+  return TRUE;
+}
+
+static gboolean
+_py_set_persist_name(PythonSourceDriver *self)
+{
+  PythonPersistMembers options =
+  {
+    .generate_persist_name_method = self->py.generate_persist_name,
+    .options = self->options,
+    .class = self->class,
+    .id = self->super.super.super.id
+  };
+
+  const gchar *persist_name = python_format_persist_name((LogPipe *)self, "python-source", &options);
```

Comment:
Same as with python-dest, consider using `python_source_format_persist_name`

Reply:
Added a refactor commit to the top. Thanks.